File: repos_estudiantes_ALVAREZ_primera_entrega/CARRERA_POZO_ALEJANDRA/Polimorfismo/Ejercicio5/Celular.py
class Celular:
    def __init__(self, nroTel, dueno, espacio, ram, nroApp):
        self.nroTel = nroTel
        self.dueno = dueno
        self.espacio = espacio 
        self.ram = ram         
        self.nroApp = nroApp

    def mostrar(self):
        print(f"Dueño: {self.dueno}")
        print(f"Teléfono: {self.nroTel}")
        print(f"Espacio: {self.espacio} GB")
        print(f"RAM: {self.ram} GB")
        print(f"Número de apps: {self.nroApp}")
        print("-" * 30)
    def __add__(self, otro):
        self.nroApp += 10
        return self
    def __sub__(self, otro):
        self.espacio -= 5
        return self
    
class CelularGamer(Celular):
    def __add__(self, otro):
        self.nroApp += 20  
        return self
    def __sub__(self, otro):
        self.espacio -= 10  
        return self

if __name__ == "__main__":
    c1 = Celular("0987-111-222", "Juan Pérez", 64, 4, 30)
    c2 = CelularGamer("0987-333-444", "María López", 128, 8, 50)
    print("Antes de los operadores")
    c1.mostrar()
    c2.mostrar()
    c1 + 1  
    c1 - 1  
    c2 + 1  
    c2 - 1  
    print("Despues de los operadores")
    c1.mostrar()
    c2.mostrar()

File: repos_estudiantes_ALVAREZ_primera_entrega/ARUQUIPA_VARGAS_ANA_GLORIA/POLIMORFISMO/Ejercicio5/Celular.java
package Ejercicio5;

public class Celular {
    private String nroTel;
    private String dueño;
    private float espacio;  
    private int ram;       
    private int nroApp;

    public Celular(String nroTel, String dueño, float espacio, int ram, int nroApp) {
        this.nroTel = nroTel;
        this.dueño = dueño;
        this.espacio = espacio;
        this.ram = ram;
        this.nroApp = nroApp;
    }

    public void incrementar() {
        this.nroApp += 10;
        System.out.println("Se aumentaron 10 aplicaciones.");
    }

    public void decrementar() {
        if (this.espacio >= 5) {
            this.espacio -= 5;
            System.out.println("Se disminuyeron 5 GB de espacio.");
        } else {
            System.out.println("No hay suficiente espacio para disminuir 5 GB.");
        }
    }

    public void mostrarDatos() {
        System.out.println("Número: " + nroTel);
        System.out.println("Dueño: " + dueño);
        System.out.println("Espacio disponible: " + espacio + " GB");
        System.out.println("RAM: " + ram + " GB");
        System.out.println("Número de apps: " + nroApp);
    }
}

